oat-sa/tao-core
# Feature/cgf 23/improve error reporting during tao update

**Related to task:**  https://oat-sa.atlassian.net/browse/CGF-23
**Description:** Improve error reporting during taoUpdate
**Depend on:** https://github.com/oat-sa/generis/pull/854
**How to test:** 
*Preconditions:* 
1. execute taoUpdate, 
2. execute composer require tl-software/opsgenie-client:2.0.2
3. create conditions so that next time taoUpdate performed with an error 

*Test first acceptance criteria(by default with no config, taoUpdate behave like now)*  
run taoUpdate again. As a result, this command should work as if this PR did not exist

*Test second acceptance criteria(when OpsGenie setup, an alert is trigger in opsGenie)*  

1. go to opsGenue, go to team section and create integration
![image](https://user-images.githubusercontent.com/18699340/98229147-9b740b00-1f6a-11eb-9816-00c341468c3e.png)

2. Get api key from this integration and put to config which will be described below
![image](https://user-images.githubusercontent.com/18699340/98229271-c0687e00-1f6a-11eb-9009-f50a0e591f22.png)

3. change the alarmNotificationService.conf.php config like this:
```
return new oat\tao\model\notifications\AlarmNotificationService([
    'notifiers' => [
        [
            'class' => '\\oat\\tao\\model\\notifiers\\OpsGenieNotifier',
            'params' => ['f3409d6d-1d96-4331-9ee0-2dd07b5b764d']
        ]
    ],
    'dispatchTypes' => [
        \oat\oatbox\reporting\Report::TYPE_ERROR
    ]
]);

```

4. run taoUpdate, as a result, error alert for opsGenie should be generated
